src/marqo/s2_inference/s2_inference.py

```diff
@@ -67,14 +77,64 @@ def vectorise(model_name: str, content: Union[str, List[str]], model_properties:
         model_auth=model_auth
     )
 
+    if _marqo_inference_cache.is_enabled() and enable_cache:
+        if isinstance(content, str):
+            inference_cache_key = f"{model_cache_key}||{content}"
```

Comment:
I think making sure that cached values for different models don't clash is InferenceCache class's responsibility. You should move this logic there. InferenceCache will then require the string, but also model etc to access the cache.